helm-imenu.el

```diff
@@ -191,6 +197,28 @@ only '((foo . bar)) is needed."
                           (helm-imenu-candidates b)))
       (progress-reporter-done progress-reporter))))
 
+(defun helm-imenu-collect-sources-from-all-buffers ()
+  (let* ((lst (buffer-list))
+         (progress-reporter (make-progress-reporter
+                             "Imenu indexing buffers..." 1 (length lst))))
+    (prog1
+        (cl-loop with cur-buf = (current-buffer)
+                 for b in lst
```

Comment:
Doesn't seem like `lst` is used anywhere else, so a call to `(buffer-list)` should be sufficient.